core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala

```diff
@@ -209,8 +209,15 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
         // External Tables
         hiveTables.collect {
           case table if table.tableType == CatalogObjectType.External.toString =>
+            val tableSrcPath = table.dataSourcePath
+            val maskedSrcPath = if (tableSrcPath.toLowerCase().startsWith("s3a://") ||
+                tableSrcPath.toLowerCase().startsWith("s3://") ||
+                tableSrcPath.toLowerCase().startsWith("s3n://") ) {
+              tableSrcPath.replace(tableSrcPath.slice(tableSrcPath.indexOf("//") + 2,
+                tableSrcPath.indexOf("@")), "****:****")
+          } else tableSrcPath
             new SnappyExternalTableStats(table.entityName, table.tableType, table.schema,
-              table.shortProvider, table.externalStore, table.dataSourcePath, table.driverClass)
+              table.shortProvider, table.externalStore, maskedSrcPath, table.driverClass)
```

Comment:
Will be better to change in StoreHiveCatalog.getDataSourcePath so that both here and sys.hivetables will mask the path. Also will handle masking passwords in sourcePath, if any. Patch to mask for all cases:

```
diff --git a/core/src/main/scala/io/snappydata/sql/catalog/impl/StoreHiveCatalog.scala b/core/src/main/scala/io/snappydata/sql/catalog/impl/StoreHiveCatalog.scala
index 4921e6df9..00b8214ee 100644
--- a/core/src/main/scala/io/snappydata/sql/catalog/impl/StoreHiveCatalog.scala
+++ b/core/src/main/scala/io/snappydata/sql/catalog/impl/StoreHiveCatalog.scala
@@ -444,16 +447,19 @@ class StoreHiveCatalog extends ExternalCatalog with Logging {
       }
     }

+    private def maskPassword(s: String): String =
+      SnappyExternalCatalog.PASSWORD_MATCH.replaceAllIn(s, "xxx")
+
     private def getDataSourcePath(properties: scala.collection.Map[String, String],
         storage: CatalogStorageFormat): String = {
       properties.get("path") match {
-        case Some(p) if !p.isEmpty => p
+        case Some(p) if !p.isEmpty => maskPassword(p)
         case _ => properties.get("region.path") match { // for external GemFire connector
-          case Some(p) if !p.isEmpty => p
+          case Some(p) if !p.isEmpty => maskPassword(p)
           case _ => properties.get("url") match { // jdbc
             case Some(p) if !p.isEmpty =>
               // mask the password if present
-              val url = SnappyExternalCatalog.PASSWORD_MATCH.replaceAllIn(p, "xxx")
+              val url = maskPassword(p)
               // add dbtable if present
               properties.get(SnappyExternalCatalog.DBTABLE_PROPERTY) match {
                 case Some(d) if !d.isEmpty => s"$url; ${SnappyExternalCatalog.DBTABLE_PROPERTY}=$d"

```